command/agentproxyshared/helpers.go

```diff
@@ -0,0 +1,227 @@
+		}
+
+		// Restore anything in the persistent cache to the memory cache
+		if err := leaseCache.Restore(ctx, ps); err != nil {
+			UI.Error(fmt.Sprintf("error restoring in-memory cache from persisted file: %v", err))
+			if persistConfig.ExitOnErr {
+				return nil, "", fmt.Errorf("exiting with error as exit_on_err is set to true")
+			}
+		}
+		logger.Info("loaded memcache from persistent storage")
+
+		// Check for previous auto-auth token
+		oldTokenBytes, err := ps.GetAutoAuthToken(ctx)
+		if err != nil {
+			UI.Error(fmt.Sprintf("error in fetching previous auto-auth token: %v", err))
+			if persistConfig.ExitOnErr {
+				return nil, "", fmt.Errorf("exiting with error as exit_on_err is set to true")
+			}
+		}
+		var previousToken string
+		if len(oldTokenBytes) > 0 {
+			oldToken, err := cachememdb.Deserialize(oldTokenBytes)
+			if err != nil {
+				UI.Error(fmt.Sprintf("error in deserializing previous auto-auth token cache entryn: %v", err))
+				if persistConfig.ExitOnErr {
+					return nil, "", fmt.Errorf("exiting with error as exit_on_err is set to true")
+				}
+			}
+			previousToken = oldToken.Token
+		}
+
+		// If keep_after_import true, set persistent storage layer in
+		// leaseCache, else remove db file
+		if persistConfig.KeepAfterImport {
+			leaseCache.SetPersistentStorage(ps)
+			return ps.Close, previousToken, nil
+		} else {
+			if err := ps.Close(); err != nil {
+				UI.Warn(fmt.Sprintf("failed to close persistent cache file: %s", err))
+			}
+			dbFile := filepath.Join(persistConfig.Path, cacheboltdb.DatabaseFileName)
+			if err := os.Remove(dbFile); err != nil {
+				UI.Error(fmt.Sprintf("failed to remove persistent storage file %s: %s", dbFile, err))
+				if persistConfig.ExitOnErr {
+					return nil, "", fmt.Errorf("exiting with error as exit_on_err is set to true")
+				}
+			}
+			return nil, previousToken, nil
+		}
+	} else {
+		km, err := keymanager.NewPassthroughKeyManager(ctx, nil)
+		if err != nil {
+			return nil, "", fmt.Errorf("failed to configure persistence encryption for cache: %v", err)
+		}
+		ps, err := cacheboltdb.NewBoltStorage(&cacheboltdb.BoltStorageConfig{
+			Path:    persistConfig.Path,
+			Logger:  logger.Named("cacheboltdb"),
+			Wrapper: km.Wrapper(),
+			AAD:     aad,
+		})
+		if err != nil {
+			return nil, "", fmt.Errorf("error creating persistent cache: %v", err)
+		}
+		logger.Info("configured persistent storage", "path", persistConfig.Path)
+
+		// Stash the key material in bolt
+		token, err := km.RetrievalToken(ctx)
+		if err != nil {
+			return nil, "", fmt.Errorf("error getting persistence key: %v", err)
+		}
+		if err := ps.StoreRetrievalToken(token); err != nil {
+			return nil, "", fmt.Errorf("error setting key in persistent cache: %v", err)
+		}
+
+		leaseCache.SetPersistentStorage(ps)
+		return ps.Close, "", nil
+	}
+}
+
+// GetServiceAccountJWT reads the service account jwt from `tokenFile`. Default is
+// the default service account file path in kubernetes.
```

Comment:
```suggestion
// getServiceAccountJWT attempts to read the service account JWT from the specified token file path.
// Defaults to using the Kubernetes default service account file path if token file path is empty.
```

Casing for godoc to match unexported func (and some tweakers).

Reply:
Thanks! Good catch :)